src/stringify/stringifyCollection.ts

```diff
@@ -148,7 +148,7 @@ function stringifyFlowCollection(
   } else {
     if (!reqNewline) {
       const len = lines.reduce((sum, line) => sum + line.length + 2, 2)
-      reqNewline = len > Collection.maxFlowStringSingleLineLength
+      reqNewline = ctx.options.lineWidth !== 0 && len > ctx.options.lineWidth
```

Comment:
"Falsy" values for `lineWidth` include negative numbers. See these examples:

https://github.com/eemeli/yaml/blob/b7696fc001837a2e9d66ad78d7c04f47943daeca/src/stringify/foldFlowLines.ts#L53

https://github.com/eemeli/yaml/blob/b7696fc001837a2e9d66ad78d7c04f47943daeca/src/stringify/stringifyString.ts#L36

Reply:
Fixed by turning it into a `ctx.options.lineWidth > 0 && ...`